### PR TITLE
Socket.IO -> Traditional web sockets

### PR DESCRIPTION
**Pull request by EricRabil**
This pull-request removes Socket.IO implementations and replaces them with a faster, more lightweight event-based WebSocket implementation. The API is very straightforward:
Server -> Client requests:
```json
{"e": "your_event_name", "d": "can be anything JSON-stringifyable"}
```
Client -> Server requests:
```json
{"r": "your_request_name", "d": "can be anything JSON-stringifyable"}
```

The changes made to the server-side code is implementation-specific and provides full backwards compatibility with all use-cases made on the server code.

The `onJSON` function in the client-side web sockets has been replaced with just `on` as the web socket wrapper now subclasses an event system.

### CODE REVIEW

**Review by nullpixel — APPROVED** [review]
Looks good. has been tested on https://staging.canvas.place and is faster and more reliable than socket.io

**Comment by nullpixel** [comment]
Waiting for @AppleBetas's approval before merging.

File: app.js
Comment (EricRabil) [inline app.js]: Please no

File: util/Sockets/SocketController.js
Comment (AppleBetas) [inline util/Sockets/SocketController.js:1]: Never used

**Comment by EricRabil** [comment]
Things to do before merging:
- Restore bootbox self-hosted
- Remove isListening calls
- Remove `unhandledRejection` and `unhandledException`

File: client/js/place.js
Comment (AppleBetas) [inline client/js/place.js:218]: Is it really a good idea to send this every time there's a keydown?

File: client/js/place.js
Comment (AppleBetas) [inline client/js/place.js:225]: Is it really a good idea to send this every time there's a mouse movement?

File: client/js/place.js
Comment (AppleBetas) [inline client/js/place.js]: ^

**Review by AppleBetas — COMMENTED** [review]
Huge improvement, just a few questions

File: util/Sockets/SocketController.js
Comment (EricRabil) [reply util/Sockets/SocketController.js:1]: I removed it

File: client/js/place.js
Comment (AppleBetas) [reply client/js/place.js:218]: Clarified, never mind

File: client/js/place.js
Comment (AppleBetas) [reply client/js/place.js:225]: Clarified, never mind

**Review by AppleBetas — APPROVED** [review]

**Comment by deadbaed** [comment]
good boiz

**Pull request merged**
